### DISCUSSION TITLE
What should MICROPYPATH be for Unix port?

### DISCUSSION THREAD

**Discussion by strnbrg59** [micropython/micropython#11779] [Unix & Windows] [answered]
I built the Unix port (on Ubuntu Linux) and immediately found that without MICROPYTHONHOME defined at all, even "import machine" or "import pyb" fail.
After setting it to $MICROPYTHONHOME/lib/micropython-lib/unix-ffi:$MICROPYTHONHOME/lib/micropython-lib
(where $MICROPYTHONHOME is the base directory where I cloned the repository and did the build)
I can import some modules (pyb, machine...) but they seem devoid of content.  ">>> import machine" followed by ">>> dir(machine)" prints simply ['__class__', '__name__', '__dict__', '__path__']
So where is everything?

On the Micropython running on my Raspberry Pi Pico, I see sys.path is ['', '.frozen', '/lib'] and yet everything works great.

I ran the tests ("make tests") and all the ones that ran succeeded.  I see lots of example code too (for example examples/network/http_server.py) but they all start having assumed you've already successfully imported modules (like network) that I am so far unable to import.

**Comment by dlech** [comment, marked as answer]
The unix port doesn't have `machine` or `pyb` modules since it generic and doesn't target any specific OS or hardware.

**Comment by strnbrg59** [comment]
OK, makes sense.  I'll have to write some stubs then; the idea is to test (to the extent possible) on Linux.
Nonetheless, if I want to benefit as much as possible from the libraries that do exist (i.e. write no more stubs than I absolutely have to), what should my MICROPYPATH be?

**Reply by dlech** [reply]
You should not have to set it to anything unless you want to import files not in the default `sys.path`, i.e. set it to the path of your stubs/mocks if you want to share those between multiple projects.

**Reply by jimmo** [reply]
@strnbrg59 MICROPYPATH defaults to `".frozen:~/.micropython/lib:/usr/lib/micropython"` which corresponds to:

```
>>> sys.path
['', '.frozen', '/home/jimmo/.micropython/lib', '/usr/lib/micropython']
```

This is a sensible default which gives you the expected behavior of finding modules in the current directory, ability to load frozen modules (e.g. uasyncio, mip), and library packages (e.g. installed by `micropython -m mip install <foo>`).
